### DISCUSSION TITLE
Can't open connection to https://dir.xiph.org and download xml file /yp.xml System.ComponentModel.Win32Exception (0x80090331): The client and server cannot communicate, because they do not possess a common algorithm.

### DISCUSSION THREAD

**Discussion by borrelnoten** [ZitchCode/SecureHttpClient#26] [Q&A] [answered]
Hi,

thank you for building this package! I'm trying to use it and download a 4Mb xml file from this ICE cast server:

https://dir.xiph.org/yp.xml

I've checked and they only support TLS 1.3 it seems.
Tried using postman and that works. It shows me the payload after downloading the xml file.
Also tried using Edge browser, no problem. 

I'm running windows 10 Pro and using .Net 9.0 in visual studio 2022.
The error I keep getting:

Exception thrown: 'System.Net.Http.HttpRequestException' in System.Private.CoreLib.dll

The SSL connection could not be established, see inner exception. System.Security.Authentication.AuthenticationException: Authentication failed, see inner exception.
 ---> System.ComponentModel.Win32Exception (0x80090331): The client and server cannot communicate, because they do not possess a common algorithm.
 
 I've configfured the httphandler like this:
 
             {
                ClientCertificateOptions = ClientCertificateOption.Automatic,
                SslProtocols = SslProtocols.Tls13, // SslProtocols.None, // SslProtocols.Tls12 |
                UseDefaultCredentials = false,
                PreAuthenticate = false,
                AutomaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate | DecompressionMethods.Brotli,
                AllowAutoRedirect = true,
                UseCookies = true,
                ServerCertificateCustomValidationCallback = (sender, cert, chain, sslPolicyErrors) => { return true; }
            };
            
Method is get via the httpclient,  and sending the same headers as postman:
            
             client.DefaultRequestHeaders.Add("Accept-Encoding", "deflate, gzip;q=1.0, *;q=0.5"); 
             client.DefaultRequestHeaders.Add("Cache-Control", "max-age=0");
client.DefaultRequestHeaders.Add("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7");
client.DefaultRequestHeaders.Add("User-Agent", "PostmanRuntime/7.43.0");
client.DefaultRequestHeaders.Add("Connection", "keep-alive");
             
I hope you can help me get me on the right track.

**Comment by tranb3r** [comment]
Are you sure TLS 1.3 is enabled on your OS ? Which cipher suite is the server using ? Is there a corresponding cipher supported by your OS ?
I don't think this issue is related to SecureHttpClient. It's probably better to ask for help on StackOverflow.
I'm converting this issue to a discussion for now. But I'm very interested in knowing about the solution.

**Comment by borrelnoten** [comment]
Thank you for your reply!
I think the client machine is not the issue, otherwise Postman and Edge browser would not return any results.
OS I'm using is windows 10 Pro running latest updates. Also updated my solution to .Net 9.0 so 1.3 is supported I think.
On the other question:  **Cipher Suites**

# TLS 1.3 (server has no preference)
TLS_AES_128_GCM_SHA256 (0x1301)   ECDH x25519 (eq. 3072 bits RSA)   FS | 128
TLS_AES_256_GCM_SHA384 (0x1302)   ECDH x25519 (eq. 3072 bits RSA)   FS | 256
TLS_CHACHA20_POLY1305_SHA256 (0x1303)   ECDH x25519 (eq. 3072 bits RSA)   FS

I checked using https://www.ssllabs.com/ssltest/analyze.html?d=dir.xiph.org

**Reply by borrelnoten** [reply, marked as answer]
I've done more research and you are 100% correct!

Windows 10 (any version) does not support .Net (any version) TLS 1.3

I've enabled it via the registry, via Edge browser (the experimantal TLS 1.3). I even tried installing the used cypher TLS_CHACHA20_POLY1305_SHA256.

Nothing works. A workaround is to make use of a "non dot net" TLS 1.3 supporting HTTPclient OR install windows 11 :)

After installing windows 11, TLS 1.3 was supported AND working. MS likes us to move to Windows 11 it seems!
